### PR TITLE
Team 7 - Adds Boss attacks

### PR DESCRIPTION
**Pull request by ThivanW**
adds attacks for the ranged boss

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java [hunk 1]
```diff
@@ -34,8 +34,14 @@ public class ForestGameArea extends GameArea {
...
+
+
+  private static final int NUM_BOSSKING2=3;
+  private static final int NUM_BOSSKING1=1;
+
+  private int wave = 0;
+  private Timer waveTimer;
+
   private Timer bossSpawnTimer;
   private int bossSpawnInterval = 10000; // 1 minute in milliseconds
```
Comment (The-AhmadAA) [inline source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:46]: This is not one minute, this is 10 seconds

File: source/core/src/main/com/csse3200/game/components/tasks/MobDeathTask.java
Comment (The-AhmadAA) [inline source/core/src/main/com/csse3200/game/components/tasks/MobDeathTask.java:1]: I thought this file was deleted, as the mob team don't use it any more. If this is for Bosses, please rename.

Comment (karthikeya-v) [reply source/core/src/main/com/csse3200/game/components/tasks/MobDeathTask.java:1]: no there is a different one bossDeathTask

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java (hunk 1 above)
Comment (karthikeya-v) [reply source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:46]: we didnt use that variable it was for a experimental thing

File: source/core/src/main/com/csse3200/game/entities/Entity.java [hunk 2]
```diff
@@ -129,10 +129,12 @@ public void setScale(Vector2 scale) {
    *
    * @param x width in metres
    * @param y height in metres
+   * @return
    */
-  public void setScale(float x, float y) {
+  public Entity setScale(float x, float y) {
```
Comment (The-AhmadAA) [inline source/core/src/main/com/csse3200/game/entities/Entity.java:134]: Why is the entity class being modified? Will this affect other teams?

**Review by The-AhmadAA — COMMENTED** [review]
Will need someone else to review before merging.

File: source/core/src/main/com/csse3200/game/entities/Entity.java (hunk 2 above)
Comment (karthikeya-v) [reply source/core/src/main/com/csse3200/game/entities/Entity.java:134]: i dont think so its returning null anyway

**Review by The-AhmadAA — APPROVED** [review]

**Comment by The-AhmadAA** [comment]
Please in future sprints consider the reviewers. Some of your code is difficult to read due to a lack of whitespace between functions, and don't use numbers in your class names

**Pull request merged**